src/cosmos/ControlledSwap.fixture.tsx

```diff
@@ -0,0 +1,164 @@
+  const disabledOutputToken: Currency | undefined = outputTokenName ? currencies[outputTokenName] : undefined
+
+  const [inputToken, setInputToken] = useState<Currency | undefined>(undefined)
+  // if (it !== inputToken) setInputToken(it)
+  const [outputToken, setOutputToken] = useState<Currency | undefined>(undefined)
+  // if (ot !== inputToken) setOutputToken(ot)
+
+  const onInputTokenChange = (curr: Currency) => {
+    console.log('onInputTokenChange', curr)
+    setInputToken(curr)
+  }
+  const onOutputTokenChange = (curr: Currency) => {
+    console.log('onOutputTokenChange', curr)
+    setOutputToken(curr)
+  }
+
+  const [amount, setAmount] = useState<string | number | undefined>(undefined)
+  const [amt] = useValue('amount', { defaultValue: 1 })
+  useEffect(() => {
+    if (amt !== amount) setAmount(amt)
+  }, [amt, amount])
+  const onAmountChange = (amount: string | number) => {
+    console.log('onAmountChange', amount)
+    setAmount(amount)
+  }
+
+  const [independentField, setIndependentField] = useState<Field | undefined>(undefined)
+  const indptField = useOption<Field>('independentField', {
+    options: [Field.INPUT, Field.OUTPUT],
+    defaultValue: Field.INPUT,
+  })
+  useEffect(() => {
+    if (indptField !== independentField) setIndependentField(indptField)
+  }, [indptField, independentField])
+  const onIndependentFieldChange = (f: Field) => {
+    console.log('onIndependentFieldChange', f)
+    setIndependentField(f)
+  }
+
+  return (
+    <SwapWidget
+      convenienceFee={convenienceFee}
+      convenienceFeeRecipient={convenienceFeeRecipient}
+      defaultChainId={defaultChainId}
+      hideConnectionUI={hideConnectionUI}
+      locale={locale}
+      jsonRpcUrlMap={jsonRpcUrlMap}
+      provider={connector}
+      theme={theme}
+      tokenList={tokenList}
+      width={width}
+      routerUrl={routerUrl}
+      onConnectWalletClick={() =>
+        new Promise((resolve) => {
+          console.log('integrator provided a onConnectWalletClick')
+          resolve(true) // to open our built-in wallet connect flow
+        })
+      }
+      onReviewSwapClick={() => new Promise((resolve) => resolve(true))}
+      onTokenSelectorClick={(field: Field) => {
+        console.log('onTokenSelectorClick')
+        return new Promise((resolve) => resolve(true))
+      }}
+      onTxSubmit={(txHash: string, data: any) => console.log('tx submitted:', txHash, data)}
+      onTxSuccess={(txHash: string, data: any) => console.log('tx success:', txHash, data)}
+      onTxFail={(error: Error, data: any) => console.log('tx fail:', error, data)}
+      // controlled token state
+      // defaultTokenSelectorDisabled={false}
+      // inputToken={inputToken}
+      // onInputTokenChange={onInputTokenChange}
+      // outputToken={outputToken}
+      // onOutputTokenChange={onOutputTokenChange}
+      defaultTokenSelectorDisabled={true}
+      inputToken={disabledInputToken}
+      outputToken={disabledOutputToken}
```

Comment:
sorry that this is so confusing -- 

if we're using the native token selectors, we want to use the token `onChange` functions to set swap state instead of setting internal token states directly.
i.e. onChange calls a setState, which we pass back into the value prop

else if we're disabling native token selectors, we want to directly use the input/outputToken props to set internal widget state.

Reply:
so, very WIP, going to need to rename these variables - but I have a 

in/outputToken, onIn/OutputTokenChange (for native defaultTokenSelectorDisabled={false})

and `disabledIn/OutputToken` (for external defaultTokenSelectorDisabled={true})